web/src/search_pill.ts

```diff
@@ -247,7 +333,7 @@ function get_search_operand(item: SearchPill, for_display: boolean): string {
     if (item.type === "search_user") {
         return item.users.map((user) => user.email).join(",");
     }
-    if (for_display && item.operator === "channel") {
+    if (for_display && item.operator === "channel" && item.operand !== "") {
```

Comment:
What's an example situation where the operand would be `""`? 

If you're just trying to make it `return item.operand`, you should do an `if` statement inside this block rather than having it fall through.

I would also consider making function do `if (!for_display) { return item.operand }` rather than checking it in each place; this function is just hard to read. Doing that refactor would also encourage adding a comment for what `for_display` means and thus why it being false means we shouldn't mutate the operand.

Reply:
the operand is `""` when we're suggesting `channel:`